diff --git a/core/modules/ckeditor5/js/ckeditor5.dialog.fix.js b/core/modules/ckeditor5/js/ckeditor5.dialog.fix.js
index 1f4ed1d7feb523cffee35f245b7dab06da4f38fa..41449dcf693fab7ff15220fd4b57bd89a71cab8d 100644
--- a/core/modules/ckeditor5/js/ckeditor5.dialog.fix.js
+++ b/core/modules/ckeditor5/js/ckeditor5.dialog.fix.js
@@ -12,6 +12,12 @@
     // CKEditor 5 in modals can work as expected.
     // @see https://api.jqueryui.com/dialog/#method-_allowInteraction
     _allowInteraction(event) {
+      // Fixes "Uncaught TypeError: event.target.classList is undefined"
+      // in Firefox (only).
+      // @see https://www.drupal.org/project/drupal/issues/3351600
+      if (event.target.classList === undefined) {
+        return this._super(event);
+      }
       return event.target.classList.contains('ck') || this._super(event);
     },
   });
