diff --git a/core/modules/system/src/Controller/AssetControllerBase.php b/core/modules/system/src/Controller/AssetControllerBase.php
index 620b2ecccf..cce4c397cc 100644
--- a/core/modules/system/src/Controller/AssetControllerBase.php
+++ b/core/modules/system/src/Controller/AssetControllerBase.php
@@ -191,10 +191,6 @@ public function deliver(Request $request, string $file_name) {
     // referenced in cached HTML.
     if (hash_equals($generated_hash, $received_hash)) {
       $uri = $this->dumper->dumpToUri($data, $this->assetType, $uri);
-      $state_key = 'drupal_' . $this->assetType . '_cache_files';
-      $files = $this->state()->get($state_key, []);
-      $files[] = $uri;
-      $this->state()->set($state_key, $files);
     }
     return new Response($data, 200, [
       'Cache-control' => static::CACHE_CONTROL,
