diff --git a/ocha_key_figures.install b/ocha_key_figures.install
new file mode 100644
index 0000000..5d7e97a
--- /dev/null
+++ b/ocha_key_figures.install
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @file
+ * Themes and preprocessors for OCHA Key Figures.
+ */
+
+use Drupal\Core\Entity\ContentEntityBase;
+
+/**
+ * Convert id to figure_id.
+ */
+function ocha_key_figures_update_9001() {
+  $field_storage_config_storage = \Drupal::entityTypeManager()->getStorage('field_storage_config');
+  $fields = $field_storage_config_storage->loadByProperties([
+    'type' => 'key_figure',
+  ]);
+
+  /** @var \Drupal\field\Entity\FieldStorageConfig $field */
+  foreach ($fields as $field) {
+    // Load all entities.
+    $entities = \Drupal::entityTypeManager()->getStorage($field->getTargetEntityTypeId())->loadMultiple();
+    foreach ($entities as $entity) {
+      if ($entity instanceof ContentEntityBase) {
+        if (!$entity->hasField($field->getName())) {
+          continue;
+        }
+
+        $values = $entity->get($field->getName())->getValue();
+        print_r($values);
+        foreach ($values as &$value) {
+          if (!isset($value['id'])) {
+            continue;
+          }
+
+          if (strpos($value['id'], '_') === FALSE) {
+            continue;
+          }
+
+          $v = $value['id'];
+          $v = explode('_', $v);
+          $value['id'] = array_pop($v);
+        }
+
+        $entity->set($field->getName(), $values);
+        $entity->save();
+      }
+    }
+  }
+}
diff --git a/src/Controller/OchaKeyFiguresController.php b/src/Controller/OchaKeyFiguresController.php
index 66f1c2b..e858d75 100644
--- a/src/Controller/OchaKeyFiguresController.php
+++ b/src/Controller/OchaKeyFiguresController.php
@@ -276,6 +276,41 @@ class OchaKeyFiguresController implements ContainerInjectionInterface {
     return $data;
   }
 
+  /**
+   * Get figure by figure id.
+   */
+  public function getFigureByFigureId(string $provider, string $country, string $year, string $figure_id) : array {
+    $query = [
+      'iso3' => $country,
+      'year' => $year,
+      'archived' => 0,
+      'figure_id' => $figure_id,
+      'order' => [
+        'year' => 'desc',
+      ],
+    ];
+
+    if ($year == 2) {
+      unset($query['year']);
+      // Get current and previous year.
+      $query['year'] = [
+        date('Y'),
+        date('Y') - 1,
+      ];
+    }
+
+    $data = $this->query($provider, '', $query);
+    if (!$data) {
+      return [];
+    }
+
+    // Keep first record.
+    $data = reset($data);
+    $data['cache_tags'] = $this->getCacheTags($data);
+
+    return $data;
+  }
+
   /**
    * Get the figures available for the figure provider, country and year.
    *
@@ -1087,7 +1122,6 @@ class OchaKeyFiguresController implements ContainerInjectionInterface {
   public function getExternalLookup(string $provider) {
     $options = [];
 
-    $prefix = $this->getPrefix($provider);
     $query = [
       'provider' => $provider,
     ];
diff --git a/src/Plugin/Field/FieldFormatter/KeyFigureBase.php b/src/Plugin/Field/FieldFormatter/KeyFigureBase.php
index 4de34ca..6f1fd0f 100644
--- a/src/Plugin/Field/FieldFormatter/KeyFigureBase.php
+++ b/src/Plugin/Field/FieldFormatter/KeyFigureBase.php
@@ -155,7 +155,7 @@ class KeyFigureBase extends FormatterBase {
     $data = $this->ochaKeyFiguresApiClient->getFigures($provider, $country, $year);
     if (!empty($data)) {
       foreach ($data as $item) {
-        $figures[$item['id']] = $item;
+        $figures[$item['figure_id']] = $item;
       }
     }
 
diff --git a/src/Plugin/Field/FieldFormatter/KeyFigureCondensed.php b/src/Plugin/Field/FieldFormatter/KeyFigureCondensed.php
index 9cf3399..b6fb58a 100644
--- a/src/Plugin/Field/FieldFormatter/KeyFigureCondensed.php
+++ b/src/Plugin/Field/FieldFormatter/KeyFigureCondensed.php
@@ -89,7 +89,7 @@ class KeyFigureCondensed extends KeyFigureBase {
 
         $data = [];
         if ($item->getFigureProvider() != 'manual') {
-          $data = $this->ochaKeyFiguresApiClient->getFigure($item->getFigureProvider(), strtolower($item->getFigureId()));
+          $data = $this->ochaKeyFiguresApiClient->getFigureByFigureId($item->getFigureProvider(), $item->getFigureCountry(), $item->getFigureYear(), strtolower($item->getFigureId()));
 
           if (isset($data['value'], $data['value_type'])) {
             $value = $data['value'];
diff --git a/src/Plugin/Field/FieldWidget/KeyFigureBaseWidget.php b/src/Plugin/Field/FieldWidget/KeyFigureBaseWidget.php
index d3501c8..278f4d0 100644
--- a/src/Plugin/Field/FieldWidget/KeyFigureBaseWidget.php
+++ b/src/Plugin/Field/FieldWidget/KeyFigureBaseWidget.php
@@ -250,7 +250,7 @@ abstract class KeyFigureBaseWidget extends WidgetBase {
     $figures = [];
     if (!empty($data)) {
       foreach ($data as $item) {
-        $figures[$item['id']] = $item;
+        $figures[$item['figure_id']] = $item;
       }
     }
 
