diff --git a/src/Controller/WebhookController.php b/src/Controller/WebhookController.php
index 991c545..481643b 100644
--- a/src/Controller/WebhookController.php
+++ b/src/Controller/WebhookController.php
@@ -2,14 +2,13 @@
 
 namespace Drupal\ocha_key_figures\Controller;
 
-use Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher;
 use Drupal\Core\Cache\Cache;
 use Drupal\Core\Controller\ControllerBase;
 use Drupal\Core\Entity\EntityFieldManager;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Logger\LoggerChannelFactoryInterface;
-use Drupal\ocha_key_figures\Controller\OchaKeyFiguresController;
 use Drupal\ocha_key_figures\Event\KeyFiguresUpdated;
+use Symfony\Component\EventDispatcher\EventDispatcher;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
@@ -43,7 +42,7 @@ class WebhookController extends ControllerBase {
   /**
    * The event dispatcher.
    *
-   * @var \Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher
+   * @var \Symfony\Component\EventDispatcher\EventDispatcher
    */
   protected $eventDispatcher;
 
@@ -61,9 +60,9 @@ class WebhookController extends ControllerBase {
     OchaKeyFiguresController $ocha_key_figure_api_client,
     EntityTypeManagerInterface $entity_type_manager,
     EntityFieldManager $entity_field_manager,
-    ContainerAwareEventDispatcher $event_dispatcher,
+    EventDispatcher $event_dispatcher,
     LoggerChannelFactoryInterface $logger_factory,
-    ) {
+  ) {
     $this->ochaKeyFiguresApiClient = $ocha_key_figure_api_client;
     $this->entityTypeManager = $entity_type_manager;
     $this->entityFieldManager = $entity_field_manager;
