diff --git a/src/AliasCleaner.php b/src/AliasCleaner.php
index 20cdb6d..bbd0a7f 100644
--- a/src/AliasCleaner.php
+++ b/src/AliasCleaner.php
@@ -70,6 +70,13 @@ class AliasCleaner implements AliasCleanerInterface {
    */
   protected $moduleHandler;
 
+  /**
+   * List of punctuation characters.
+   *
+   * @var array
+   */
+  protected $punctuationCharacters;
+
   /**
    * Creates a new AliasCleaner.
    *
