diff --git a/src/Hook/UsernameEnumerationPreventionHooks.php b/src/Hook/UsernameEnumerationPreventionHooks.php
new file mode 100644
index 0000000..25fbf14
--- /dev/null
+++ b/src/Hook/UsernameEnumerationPreventionHooks.php
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\username_enumeration_prevention\Hook;
+
+use Drupal\Core\Asset\AttachedAssetsInterface;
+use Drupal\Core\Hook\Attribute\Hook;
+use Drupal\Core\Hook\Order\Order;
+use Drupal\Core\Routing\CurrentRouteMatch;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+/**
+ * Hook implementations for username_enumeration_prevention.
+ */
+readonly class UsernameEnumerationPreventionHooks {
+
+  /**
+   * The UsernameEnumerationPreventionHooks constructor.
+   *
+   * @param \Drupal\Core\Routing\CurrentRouteMatch $routeMatch
+   *   The route match.
+   */
+  public function __construct(
+    #[Autowire('@current_route_match')]
+    private CurrentRouteMatch $routeMatch,
+  ) {
+  }
+
+  /**
+   * Implements hook_js_settings_alter().
+   */
+  #[Hook('js_settings_alter', order: Order::Last)]
+  public function jsSettingsAlter(array &$settings, AttachedAssetsInterface $assets): void {
+    if ($this->routeMatch->getRouteName() === 'system.404') {
+      $settings['path']['currentPath'] = '';
+    }
+  }
+
+}
