diff --git a/config/install/user_expire.settings.yml b/config/install/user_expire.settings.yml
index 62f9e8c7d5a064d87448b0faf5811a08709370c7..ea9fb144d2add8031bcafc516fb46cca209eb6c9 100644
--- a/config/install/user_expire.settings.yml
+++ b/config/install/user_expire.settings.yml
@@ -1,3 +1,7 @@
 frequency: 172800
 offset: 604800
 user_expire_roles: {}
+send_mail: true
+mail:
+  subject: "[site:name]: Account expiration warning"
+  body: "Hello [user:display-name]\r\n\r\nBecause you have not logged in recently, your account at [site:name] will be blocked in the near future. If you still use this site, please log in [site:login-url] to prevent your account being blocked.\r\n\r\n-- [site:name] team"
diff --git a/config/schema/user_expire.schema.yml b/config/schema/user_expire.schema.yml
index a46bc68eeea8b41fa1e7bde72008daf77d54032d..be0c3fca4e7604994e5c0070942ef79920eab454 100644
--- a/config/schema/user_expire.schema.yml
+++ b/config/schema/user_expire.schema.yml
@@ -7,9 +7,21 @@ user_expire.settings:
     frequency:
       type: integer
       label: 'Frequency time in seconds'
+    mail:
+      type: mapping
+      mapping:
+        subject:
+          type: string
+          label: 'Subject line for the notification email'
+        body:
+          type: string
+          label: 'Body for the notification email'
     offset:
       type: integer
       label: 'Warning offset time in seconds'
+    send_mail:
+      type: boolean
+      label: 'Flag that enables or disables expiry emails'
     user_expire_roles:
       type: sequence
       label: 'Roles and expire value'
diff --git a/src/Form/UserExpireSettingsForm.php b/src/Form/UserExpireSettingsForm.php
index f7824c09ad10612818c2cc3e2cc2a85e590377a7..cd2c4e515d0887df9a70fe0c7c72bc599de84033 100644
--- a/src/Form/UserExpireSettingsForm.php
+++ b/src/Form/UserExpireSettingsForm.php
@@ -109,6 +109,50 @@ class UserExpireSettingsForm extends ConfigFormBase {
       ];
     }
 
+    // Enable or disable email notifications.
+    $form['send_mail'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Send notification emails'),
+      '#default_value' => $config->get('send_mail') ?: TRUE,
+      '#description' => $this->t('Send a notification email to the user, starting at the defined offset time before account expiry.'),
+    ];
+
+    // Notification email template.
+    $form['mail'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Notification email'),
+    ];
+
+    $form['mail']['settings'] = [
+      '#type' => 'container',
+      '#states' => [
+        // Hide the additional settings when this email is disabled.
+        'invisible' => [
+          'input[name="send_mail"]' => ['checked' => FALSE],
+        ],
+      ],
+    ];
+
+    $form['mail']['settings']['notification_subject'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Subject'),
+      '#default_value' => $config->get('mail.subject') ?: '',
+      '#description' => $this->t('Subject line for the notification email.'),
+      '#maxlength' => 180,
+    ];
+
+    $form['mail']['settings']['notification_body'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Body'),
+      '#default_value' => $config->get('mail.body') ?: '',
+      '#description' => $this->t('Body for the notification email.'),
+      '#rows' => 15,
+    ];
+
+    $form['mail']['settings']['help'] = [
+      '#markup' => $this->t('Available token variables for use in the email are: [site:name], [site:url], [site:mail], [user:display-name], [user:account-name], [user:mail], [site:login-url], [site:url-brief], [user:edit-url], [user:one-time-login-url], [user:cancel-url]'),
+    ];
+
     return parent::buildForm($form, $form_state);
   }
 
@@ -160,6 +204,13 @@ class UserExpireSettingsForm extends ConfigFormBase {
     }
 
     $config->set('user_expire_roles', $rules);
+
+    // The notification email.
+    $config->set('send_mail', $form_state->getValue('send_mail'));
+
+    $config->set('mail.subject', $form_state->getValue('notification_subject'));
+    $config->set('mail.body', $form_state->getValue('notification_body'));
+
     $config->save();
   }
 
diff --git a/user_expire.module b/user_expire.module
index d4dcd8ba28b51ee8f54a7626d8b10443244520d9..9365c5db3b87e88dd03611a03656370e2b4f99cb 100644
--- a/user_expire.module
+++ b/user_expire.module
@@ -5,6 +5,7 @@
  * Main module file for User expire module.
  */
 
+use Drupal\Component\Render\PlainTextOutput;
 use Drupal\Core\Database\Query\Condition;
 use Drupal\Core\Database\StatementInterface;
 use Drupal\Core\Datetime\DrupalDateTime;
@@ -332,12 +333,15 @@ function user_expire_expire_by_role_warning(): void {
           ]);
         }
         else {
-          $logger->info('Warning about expiring account @name by role', ['@name' => $account->getAccountName()]);
-          \Drupal::service('plugin.manager.mail')->mail('user_expire', 'expiration_warning', $account->getEmail(), $account->getPreferredLangcode(),
-            [
-              'account' => $account,
-            ]
-          );
+          // Send a notification email if configured to do so.
+          if ($config->get('send_mail')) {
+            $logger->info('Sending warning about expiring account @name by role', ['@name' => $account->getAccountName()]);
+            \Drupal::service('plugin.manager.mail')->mail('user_expire', 'expiration_warning', $account->getEmail(), $account->getPreferredLangcode(),
+              [
+                'account' => $account,
+              ]
+            );
+          }
         }
       }
     }
@@ -439,20 +443,22 @@ function user_expire_get_role_rules(): mixed {
  */
 function user_expire_mail($key, &$message, $params): void {
   if ($key == 'expiration_warning') {
-    $site_name = \Drupal::config('system.site')->get('name');
-    // The subject.
-    $message['subject'] = t('@site_name: Account expiration warning', ['@site_name' => $site_name]);
-    // The body.
-    $message['body'][] = t('Hello @user', ['@user' => $params['account']->getAccountName()]);
-    // An empty string gives a newline.
-    $message['body'][] = '';
-    $message['body'][] = t('Because you have not logged in recently, your account at @site_name will be blocked in the near future. If you still use this site, please log in @login_url to avoid having your account blocked.',
-      [
-        '@site_name' => $site_name,
-        '@login_url' => Url::fromRoute('entity.user.canonical', ['user' => \Drupal::currentUser()->id()], ['absolute' => TRUE])->toString(),
-      ]
-    );
-    $message['body'][] = '';
-    $message['body'][] = t('Thanks, @site_name', ['@site_name' => $site_name]);
+    $token_service = \Drupal::token();
+    $language_manager = \Drupal::languageManager();
+    $langcode = $message['langcode'];
+    $variables = ['user' => $params['account']];
+
+    $language = $language_manager->getLanguage($params['account']->getPreferredLangcode());
+    $original_language = $language_manager->getConfigOverrideLanguage();
+    $language_manager->setConfigOverrideLanguage($language);
+
+    $config_factory = \Drupal::configFactory();
+    $config = $config_factory->get('user_expire.settings');
+
+    $token_options = ['langcode' => $langcode, 'callback' => 'user_mail_tokens', 'clear' => TRUE];
+    $message['subject'] .= PlainTextOutput::renderFromHtml($token_service->replace($config->get('mail.subject'), $variables, $token_options));
+    $message['body'][] = $token_service->replace($config->get('mail.body'), $variables, $token_options);
+
+    $language_manager->setConfigOverrideLanguage($original_language);
   }
 }
