diff --git a/tests/src/Functional/UserExpireTest.php b/tests/src/Functional/UserExpireTest.php
index c746350f47259d17d458e4cfdee81e541ea55062..b427dfee1793e26094154573375f66927b56e7be 100644
--- a/tests/src/Functional/UserExpireTest.php
+++ b/tests/src/Functional/UserExpireTest.php
@@ -167,6 +167,18 @@ class UserExpireTest extends BrowserTestBase {
     // Ensure they are disabled.
     $this->drupalGet("user/" . $new_basic_account->id() . "/edit");
     $this->assertSession()->responseContains('type="radio" id="edit-status-0" name="status" value="0" checked="checked" class="form-radio"', $this->t('User account is currently disabled.'));
+
+    // Manually unblock the user.
+    $edit = [];
+    $edit['status'] = 1;
+    $this->submitForm($edit, $this->t('Save'));
+
+    // Process it.
+    user_expire_expire_by_role();
+
+    // Ensure they are still active.
+    $this->drupalGet("user/" . $new_basic_account->id() . "/edit");
+    $this->assertSession()->responseContains('type="radio" id="edit-status-1" name="status" value="1" checked="checked" class="form-radio"', $this->t('User account is currently active.'));
   }
 
 }
diff --git a/user_expire.module b/user_expire.module
index d4dcd8ba28b51ee8f54a7626d8b10443244520d9..fae1de17d4bb1b0a1a3819446c65817daaccf424 100644
--- a/user_expire.module
+++ b/user_expire.module
@@ -12,6 +12,7 @@ use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Routing\RouteMatchInterface;
 use Drupal\Core\Url;
+use Drupal\user\Entity\User;
 use Drupal\user\RoleInterface;
 
 /**
@@ -456,3 +457,15 @@ function user_expire_mail($key, &$message, $params): void {
     $message['body'][] = t('Thanks, @site_name', ['@site_name' => $site_name]);
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_presave() for user entities.
+ *
+ * If the account was blocked but is now active, update the expiry so it is
+ * not re-blocked by the next cron run.
+ */
+function user_expire_user_presave(User $account) {
+  if (!empty($account->original) && $account->original->isBlocked() && $account->isActive()) {
+    $account->setLastAccessTime(\Drupal::time()->getRequestTime());
+  }
+}
